base/inference.jl

```diff
@@ -3365,40 +3460,44 @@ function optimize(me::InferenceState)
     type_annotate!(me)
 
     # run optimization passes on fulltree
-    force_noinline = false
+    force_noinline = true
```

Comment:
Regardless of what the correct default is here, this **should not be included in a large, unrelated pull request**. If this is wrong, it should be changed in a separate individual PR with a commit message that explains what was wrong and why the change is correct.